package/MDAnalysis/transformations/boxdimensions.py

```diff
@@ -80,15 +99,20 @@ def __init__(self,
                        'np.float32 numpy.ndarray')
             raise ValueError(errmsg)
         try:
-            self.dimensions = self.dimensions.reshape(6, )
+            self.dimensions = self.dimensions.reshape(-1, 6)
         except ValueError:
             errmsg = (f'{self.dimensions} array does not have valid box '
                        'dimension shape.\nSimulation box dimensions are '
-                       'given by an float array of shape (6, ), '
-                       ' containing 3 lengths and 3 angles: '
+                       'given by an float array of shape (6, 0), (1, 6), '
+                       'or (N, 6) where N is the number of frames in the '
+                       'trajectory and the dimension vector(s) containing '
+                       '3 lengths and 3 angles: '
                        '[a, b, c, alpha, beta, gamma]')
             raise ValueError(errmsg)
 
     def _transform(self, ts):
-        ts.dimensions = self.dimensions
+        try:
+            ts.dimensions = self.dimensions[0] if self.dimensions.shape[0] == 1 else self.dimensions[ts.frame]
+        except IndexError as e:
+            raise ValueError(f"Dimensions array has no data for frame {ts.frame}") from e
```

Comment:
```suggestion
            errmsg = f"Dimensions array has no data for frame {ts.frame}"
            raise ValueError(errmsg) from e
```
I know it's annoying but pep8 - note we don't care about the `black` linter stuff, but the flake8 errors from here probably should be addressed: https://github.com/MDAnalysis/mdanalysis/actions/runs/5997923144/job/16265248319

Reply:
ah sorry I forgot to run flake8 locally, fixed now! Perhaps adding a pre-commit config for people to optionally use would be useful?

Reply:
> pre-commit config for people to optionally use would be useful?

It's.. complicated, I'll leave it as that for now.

Reply:
ha! okay :)